### PR TITLE
Place Java setup in re-usable workflow file

### PR DESCRIPTION
**Pull request by jonkoops**
Creates a re-usable workflow to set up Java (similar to the NPM setup). This allow us to remove some repeated code for places were we want to use the project default Java version and distribution.

### CODE REVIEW

**Review by ahus1 — DISMISSED** [review]
Thanks for consolidating the Java setup!

File: .github/actions/unit-test-setup/action.yml [hunk 1]
```diff
@@ -1,25 +1,11 @@
 name: Setup unit test
 description: Download Maven caches needed for unit tests
 
-inputs:
```
Comment (stianst) [inline .github/actions/unit-test-setup/action.yml:4]: Should leave the inputs, and pass on to `actions/java-setup` as it should be possible to use `unit-test-setup` with a different JDK version if needed

File: .github/actions/integration-test-setup/action.yml [hunk 2]
```diff
@@ -9,14 +9,14 @@ inputs:
   jdk-version:
     description: JDK version
     required: false
-    default: 17
+    default: "17"
```
Comment (stianst) [inline .github/actions/integration-test-setup/action.yml:12]: Does this need to be quoted? I'm curious why if so.

Comment (ahus1) [reply .github/actions/integration-test-setup/action.yml:12]: The JSON schema for GitHub actions defines this as a string, and YAML will identify it as a number if it is not quoted. The same for boolean values.

**Review by stianst — APPROVED** [review]

**Pull request merged**